pkg/kubeapiserver/options/admission_test.go

```diff
@@ -19,37 +19,64 @@ package options
 import (
 	"reflect"
 	"testing"
+
+	"k8s.io/apiserver/pkg/admission"
+	genericoptions "k8s.io/apiserver/pkg/server/options"
 )
 
 func TestValidate(t *testing.T) {
-	// 1. Both `--admission-control` and `--enable-admission-plugins` are specified
-	options := NewAdmissionOptions()
-	options.PluginNames = []string{"ServiceAccount"}
-	options.GenericAdmission.EnablePlugins = []string{"Initializers"}
-	if len(options.Validate()) == 0 {
-		t.Errorf("Expect error, but got none")
-	}
-
-	// 2. Both `--admission-control` and `--disable-admission-plugins` are specified
-	options = NewAdmissionOptions()
-	options.PluginNames = []string{"ServiceAccount"}
-	options.GenericAdmission.DisablePlugins = []string{"Initializers"}
-	if len(options.Validate()) == 0 {
-		t.Errorf("Expect error, but got none")
-	}
-
-	// 3. PluginNames is not registered
-	options = NewAdmissionOptions()
-	options.PluginNames = []string{"pluginA"}
-	if len(options.Validate()) == 0 {
-		t.Errorf("Expect error, but got none")
+	tests := []struct {
+		name             string
+		pluginNames      []string
+		genericAdmission *genericoptions.AdmissionOptions
+		wantError        bool
+	}{
+		{
+			name:        "test1-Both `--admission-control` and `--enable-admission-plugins` are specified",
```

Comment:
can we remove the `test#` prefix and generate that automatically in the loop?

Reply:
Directly defined outside will be more intuitive to troubleshoot (as long as the name is unique).